### PR TITLE
Assignment3

### PR DESCRIPTION
**Pull request by Ashrom0**

### CODE REVIEW

**Review by czamora7 — COMMENTED** [review]
Create two interfaces. One to represent the JokeService and one interface to represent outputting the joke to the screen (Console.WriteLine). ✔

The JokeService will need to have the interface applied to it. ✔

Create an implementation of the output interface that writes out the joke to the console. ✔

Implement the Jester class. It should take in both interfaces as dependencies. These dependencies should be null checked. ✔

The Jester class TellJoke() method should retrieve a joke from the JokeService. If the joke contains "Chuck Norris", skip it and get another. The joke should be written to the output dependency. ✔

Unit test the Jester class. Code coverage should be above 90% for this class. Moq will make this significantly easier. ❌

Enable nullability in the .csproj files and ensure no errors or warnings: ✔

Fundamentals
Ensure you enable:
nullable reference types is enabled ✔
net6 targetted ✔
C# 10.0 ✔
and enabled .NET analyzers for both projects ✔
For this assignment, always use Assert.AreEqual<T>() (the generic version) ❌
All of the above should be unit tested ❌
Choose simplicity over complexity ✔

File: CanHazFunny/CanHazFunny.Tests/JesterTests.cs [hunk 1]
```diff
@@ -1,10 +1,68 @@
...
+        [TestMethod]
+        public void TellJoke_ChuckNorrisTest_FindNewJoke()
+        {
+
+            Mock<JokeService> mock = new Mock<JokeService>();
+
+            mock.SetupSequence(jokeService => jokeService.GetJoke())
+                .Returns("Chuck Norris does not sleep. He waits.")
+                .Returns("Chuck Norris can divide by zero, and counted to infinity 3 times.")
+                .Returns("When Chuck Norris falls in water, Chuck Norris doesn't get wet. Water gets Chuck Norris.")
```
Comment (mmwoodfo) [inline CanHazFunny/CanHazFunny.Tests/JesterTests.cs:56]:
```suggestion
            mock.SetupSequence(jokeService => jokeService.GetJoke())
                .Returns("Chuck Norris does not sleep. He waits.")
                .Returns("Chuck Norris can divide by zero, and counted to infinity 3 times.")
                .Returns("When Chuck Norris falls in water, Chuck Norris doesn't get wet. Water gets Chuck Norris.");
```

Comment (mmwoodfo) [reply CanHazFunny/CanHazFunny.Tests/JesterTests.cs:56]: This is causing your code not to compile.

File: CanHazFunny/CanHazFunny.Tests/JesterTests.cs [hunk 2]
```diff
@@ -1,10 +1,68 @@
...
+
+        [TestMethod]
+        public void Jester_WorkingJokeOutput()
+        {
+            JokeService jokeService = new JokeService();
+            JokeOutput jokeOutput = new JokeOutput();
+
+            Jester jester = new Jester(jokeService, jokeOutput);
+
+            Assert.AreEqual(jokeService, jokeOutput);
```
Comment (mmwoodfo) [inline CanHazFunny/CanHazFunny.Tests/JesterTests.cs:32]: Why check if these are equal? Shouldn't you check that this doesn't throw an exception if a null input is introduced?

File: CanHazFunny/CanHazFunny/Jester.cs [hunk 3]
```diff
@@ -0,0 +1,26 @@
+﻿using System;
+
+namespace CanHazFunny
+{
+    public class Jester
+    {
+        private JokeService jokeService;
+        private JokeOutput jokeOutput;
```
Comment (mmwoodfo) [inline CanHazFunny/CanHazFunny/Jester.cs:8]:
```suggestion
        private JokeService JokeService {get;}
        private JokeOutput JokeOutput {get;}
```

**Pull request closed without merging**